### PR TITLE
Python tools improved incoming check

### LINKED ISSUE #2844: python tools --incoming check should check if the project has no index yet
I'm not sure if this is implemented or not right now but I'm adding an issue to find this out.

When using the incoming check for mirror.py + sync.py combo (after #2843 is fixed), I suspect that for the first time the sync.py is run (projects are freshly clonned to src_root, no index exists for them), the incomming check would skip the indexing for all of the projects (=no changes as expected because it has just been clonned). So the check could be more robust and check the index presence and also the scm changes presence.

### PR DESCRIPTION
**Pull request by tulinkry**
Improving --incoming check to check also the index presence
for the particular project.
    
After that, the name "incoming" could be a little misleading
so I refactored the name to "check changes".
    
fixes #2844

### CODE REVIEW

File: opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py [hunk 1]
```diff
@@ -244,22 +246,44 @@ def mirror_project(config, project_name, check_incoming, uri,
...
+            r.raise_for_status()
+            if not bool(r.json()):
+                changes_detected = True
+                logger.debug('Project {} has not been indexed yet'
+                             .format(project_name))
+        except ValueError as e:
+            logger.error('Unable to parse project \'{}\' indexed flag: {}'
+                         .format(project_name, e))
+            return FAILURE_EXITVAL
+        except Exception as e:
```
Comment (vladak) [inline opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py]: maybe too broad ?

File: opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py [hunk 2]
```diff
@@ -244,22 +246,44 @@ def mirror_project(config, project_name, check_incoming, uri,
...
+            logger.error('Unable to determine project \'{}\' indexed flag: {}'
+                         .format(project_name, e))
+            return FAILURE_EXITVAL
 
-        if not got_incoming:
+        # check if the project has any new changes in the scm
```
Comment (vladak) [inline opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py]: SCM

File: opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py [hunk 3]
```diff
@@ -196,13 +198,13 @@ def get_project_properties(project_config, project_name, hookdir):
         use_proxy, ignored_repos
 
 
-def mirror_project(config, project_name, check_incoming, uri,
+def mirror_project(config, project_name, check_changes, uri,
```
Comment (vladak) [inline opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py]: the function is getting too long. refactoring needed.

File: opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py [hunk 4]
```diff
@@ -196,13 +198,13 @@ def get_project_properties(project_config, project_name, hookdir):
...
                    source_root):
     """
     Mirror the repositories of single project.
     :param config global configuration dictionary
     :param project_name: name of the project
-    :param check_incoming:
+    :param check_changes:
```
Comment (vladak) [inline opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py]: add some description pls

File: opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py (hunk 1 above)
Comment (tulinkry) [reply opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py]: changed to HTTPError from requests

File: opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py (hunk 4 above)
Comment (tulinkry) [reply opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py]: added

File: opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py (hunk 3 above)
Comment (tulinkry) [reply opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py]: agreed

Comment (vladak) [reply opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py]: pls file a new issue then or maybe even do it now

Comment (tulinkry) [reply opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py]: https://github.com/oracle/opengrok/issues/2849

**Pull request merged**